builder/cloudstack/ssh.go

```diff
@@ -7,28 +7,30 @@ import (
 
 	packerssh "github.com/hashicorp/packer/communicator/ssh"
 	"github.com/mitchellh/multistep"
-	"github.com/xanzy/go-cloudstack/cloudstack"
 	"golang.org/x/crypto/ssh"
 	"golang.org/x/crypto/ssh/agent"
 )
 
 func commHost(state multistep.StateBag) (string, error) {
-	client := state.Get("client").(*cloudstack.CloudStackClient)
-	config := state.Get("config").(*Config)
 
-	if config.hostAddress == "" {
-		ipAddr, _, err := client.Address.GetPublicIpAddressByID(config.PublicIPAddress)
```

Comment:
Unless I missed something this should work.

1) Using local IP it is set on [step_create_instance.go#L133](https://github.com/rickard-von-essen/packer/blob/1bf8a0c6a146481500acf939425d53a505952baf/builder/cloudstack/step_create_instance.go#L133)
2) Using a temporary public IP address and it's set on [step_configure_networking.go#L88](https://github.com/rickard-von-essen/packer/blob/1bf8a0c6a146481500acf939425d53a505952baf/builder/cloudstack/step_configure_networking.go#L88)
3) Using a configured public IP address and it's set on [step_prepare_config.go#L67](https://github.com/rickard-von-essen/packer/blob/1bf8a0c6a146481500acf939425d53a505952baf/builder/cloudstack/step_prepare_config.go#L67)